tests/golden/check-autolinks/check-autolinks.bats

```diff
@@ -27,6 +27,17 @@ assert_diff - <<EOF
     - anchors:
         none
 
-All repository links are valid.
+=== Invalid references found ===
+
+  ➥  In file file-with-autolinks.md
+     bad reference (external) at src:8:0-18:
+       - text: "www.commonmark.org"
+       - link: http://www.commonmark.org
+       - anchor: -
+
+     Permanent redirect found. Perhaps you want to update the link:
+     http://www.commonmark.org -> https://commonmark.org/
```

Comment:
Will this become a problem? HTTP requests that redirect to their respective HTTPS version.

Reply:
I think this is fine, IMO the user should provide the direct `https` link in such case so we fit the general rule.

Reply:
Even if `http` was an auto-inserted default. AFAIU if the user dislikes this behaviour, it will be configurable too.

Maybe when describing configurable redirects later in `README`, we can provide an example with suppresses errors about `http -> https`.

Reply:
So we even immediately have one test for this, neat :smile: